packages/dd-trace/src/appsec/sdk/user_blocking.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const addresses = require('../addresses')
+const Gateway = require('../gateway/engine')
+const { getRootSpan } = require('./utils')
+const { block } = require('../blocking')
+const { storage } = require('../../../../datadog-core')
+const { setUserTags } = require('./set_user')
+const log = require('../../log')
+
+function isUserBlocked (user) {
+  const results = Gateway.propagate({ [addresses.USER_ID]: user.id })
+
+  if (!results) {
+    return false
+  }
+
+  for (const entry of results) {
+    if (entry && entry.includes('block')) {
+      return true
+    }
+  }
+
+  return false
+}
+
+function checkUserAndSetUser (tracer, user) {
+  if (!user || !user.id) {
+    return false
+  }
+
+  const rootSpan = getRootSpan(tracer)
+  if (!rootSpan) {
+    log.warn('Root span not available in checkUserAndSetUser')
```

Comment:
The user doesn't know what checkUserAndSetUser is, it knows only `isUserBlocked` so i'd say we rename the log to reflect that ?